spec/std/float_printer/ryu_printf_spec.cr

```diff
@@ -0,0 +1,3187 @@
+# FIXME: this leads to an OOB on wasm32 (#13918)
+{% skip_file if flag?(:wasm32) %}
+
+# This file contains test cases derived from:
+#
+# * https://github.com/ulfjack/ryu
+#
+# The following is their license:
+#
+#   Copyright 2020-2021 Junekey Jeon
+#
+#   The contents of this file may be used under the terms of
+#   the Apache License v2.0 with LLVM Exceptions.
+#
+#      (See accompanying file LICENSE-Apache or copy at
+#       https://llvm.org/foundation/relicensing/LICENSE.txt)
+#
+#   Alternatively, the contents of this file may be used under the terms of
+#   the Boost Software License, Version 1.0.
+#      (See accompanying file LICENSE-Boost or copy at
+#       https://www.boost.org/LICENSE_1_0.txt)
+#
+#   Unless required by applicable law or agreed to in writing, this software
+#   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#   KIND, either express or implied.
+
+require "spec"
+require "float/printer/ryu_printf"
+require "big"
+
+# The following two lists store: {value, fixed_precision, exp_precision}
+# Iterate over them via macros only, otherwise the entire array literal will be
+# materialized!
+# The original specs contain also the full expected results which would have
+# increased the file size by over 18x. Here we assume that `BigFloat#to_s` and
+# `BigDecimal#to_s` are always correct, so that we don't have to write the long
+# results in the source code.
+
+# These values test every power of ten that's within the range of doubles.
+ALL_POWERS_OF_TEN = [
```

Comment:
**suggestion:** Maybe we could move these constants into separate data files to make this file more navigable? It's quite annoying having to skip thousands of lines just to get to the actual specs.

Reply:
The other option is to simply move those constants to the bottom of the file

Reply:
I'm not sure that would work because the macro would reference a constant before its defined. Might be doable somehow.
However: It sure helps making the specs directly accessible, but the file is still huge and hard to navigate. It seems very natural to separate that much data into another file.